hack/usb-iso.sh

```diff
@@ -0,0 +1,154 @@
+	echo "$USAGE"
+	exit 1
+    fi
+    # normalize device name
+    USB_DEV=$(echo ${USB_DEV} | sed -r 's|/dev/||')
+    USB_DEV=/dev/${USB_DEV}
+
+    if [[ -z "${IGNITION_URL}" ]]; then
+	echo "ignition url must be specified!"
+	echo "$USAGE"
+	exit 1
+    fi
+  
+    if [[ -z "${INSTALLER_IMAGE}" ]]; then
+        echo "installer image url must be specified!"
+	echo "$USAGE"
+	exit 1
+    fi
+
+    if [[ -z "${METAL_IMAGE}" ]]; then
+	echo "metal image url must be specified!"
+	echo "$USAGE"
+	exit 1
+    fi
+
+    BLOCK_DEV=${BLOCK_DEV:-sda}
+
+    # by default flash usb key and clean up intermediate file folders after flash
+    # otherwise don't flash and leave the intermediate files for debug
+}
+
+
+main() {
+    parse_args $@
+
+    # test device make sure it is present
+    if ! lsblk ${USB_DEV} > /dev/null && [[ ${WRITE_USB} == "true" ]]; then
+	echo "usb device must be inserted unless WRITE_USB is set to false!"
+	exit 1
+    fi
+
+    for link in ${IGNITION_URL} ${INSTALLER_IMAGE} ${METAL_IMAGE}; do
+	if ! curl -LsIf ${link} >/dev/null 2>&1; then
+            echo "invalid link: ${link}"
+	    exit 1
+	fi
+    done
+
+    /bin/rm -rf _data
+    mkdir -p _data
+    pushd _data
+    mkdir -p _extra/extra
+    echo "Downloading ignition file ..."
+    if ! curl -L -s -o _extra/extra/node.ign ${IGNITION_URL} >/dev/null 2>&1; then
+	echo "failed to get ${IGNITION_URL}!"
+	exit 1
+    fi
+    echo "Downloading installer ..."
+    if ! curl -L -s -o installer.iso ${INSTALLER_IMAGE} >/dev/null 2>&1; then
+	echo "failed to get ${INSTALLER_IMAGE}!"
+        exit 1
+    fi
+    echo "Downloading metal image ..."
+    if ! curl -L -s -o _extra/extra/rhcos.raw.gz ${METAL_IMAGE} >/dev/null 2>&1; then
+	echo "failed to get ${METAL_IMAGE}!"
+	exit 1
+    fi
+
+    # setup booting parameter
+    /bin/rm -rf _rhcos_custom
+    mkdir _rhcos_custom
+    mkdir _iso_mount
+
+    echo "mounting installer iso ..."
+    mount -t iso9660 -o loop installer.iso _iso_mount 
+
+    echo "copy out installer files ..."
+    pushd _iso_mount 
+    tar cf - . | (cd ../_rhcos_custom && tar xfp -)
+    popd
+
+    echo "update installer parameters ..."
+    pushd _rhcos_custom 
+    sed -i -r "s|^(  append initrd)=.*|\1=/images/initramfs.img,/images/initramfsExtra nomodeset ip=dhcp rd.neednet=1 coreos.inst=yes console=tty0 console=ttyS0 coreos.inst.ignition_url=file:///extra/node.ign coreos.inst.image_url=file:///extra/rhcos.raw.gz coreos.inst.install_dev=${BLOCK_DEV}|g" isolinux/isolinux.cfg
+    popd
+
+    echo "building extra initramfs ..."
+    pushd _extra
+    find . | sed 's/^[.]\///' | cpio -o -H newc --no-absolute-filenames > ../_rhcos_custom/images/initramfsExtra
+    popd
+   
+    echo "rebuilding installer image ..."
+    pushd _rhcos_custom
+    mkisofs -o rhcos_custom_usb.iso -b isolinux/isolinux.bin -c isolinux/boot.cat  -no-emul-boot -boot-load-size 4 -boot-info-table -R -J -V "RHCOS custom installer" .
```

Comment:
when burning to an usb, we need:
isohybrid rhcos_custom_usb.iso

After the iso creation, so it generates the proper structure for the usb to boot.